src/app/colors/colormap-colorizer/colormap-colorizer.component.ts

```diff
@@ -0,0 +1,128 @@
+import {ChangeDetectionStrategy, ChangeDetectorRef, Component, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+import {ColorizerData} from '../colorizer-data.model';
+import {
+    BoundedMplColormapStepScale,
+    COLORMAP_STEP_SCALES_WITH_BOUNDS,
+    MPL_COLORMAP_NAMES,
+    MplColormap,
+    MplColormapName
+} from '../mpl-colormaps/mpl-colormap.model';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {valueAboveMin, valueBelowMax, WaveValidators} from '../../util/form.validators';
+import {Subscription} from 'rxjs';
+
+@Component({
+    selector: 'wave-colormap-colorizer',
+    templateUrl: 'colormap-colorizer.component.html',
+    styleUrls: ['colormap-colorizer.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ColormapColorizerComponent implements OnInit, OnDestroy {
+
+    @Output() colormapColorizerData = new EventEmitter<ColorizerData>();
+    @Input() defaultNumberOfSteps = 16;
+    @Input() maxColormapSteps = 16;
+
+    colormapNames = MPL_COLORMAP_NAMES;
+    boundedColormapStepScales = COLORMAP_STEP_SCALES_WITH_BOUNDS;
+    form: FormGroup;
+    colorizerData: ColorizerData = MplColormap.createColorizerDataWithName(this.colormapNames[0], 0, 1);
+    subscriptions: Array<Subscription> = [];
+
+    constructor(
+        private changeDetectorRef: ChangeDetectorRef,
+        private formBuilder: FormBuilder,
+    ) {
+        this.form = formBuilder.group({
+                bounds: formBuilder.group({
+                    min: [0],
+                    max: [1]
+                }, {
+                    validators: [WaveValidators.minAndMax('min', 'max', {checkBothExist: true})]
+                }),
+                colormapName: [this.colormapNames[0], [Validators.required]],
+                colormapSteps: [this.defaultNumberOfSteps, [Validators.required, Validators.min(2)]],
+                colormapStepScales: [this.boundedColormapStepScales[0]]
+            }, {
+                validators: [
+                    valueAboveMin(
```

Comment:
wieso wird der nicht auf min und max direkt definiert?
Dann kann man sich doch den komplizierten Zugriff über den Function-Getter sparen.

Reply:
weil ich die erlaubten Werte aus dem ausgewählten colormapscales benötige. Die sind ja nicht immer gleich.

Reply:
Das ist der zweite Parameter, oder? Den verstehe ich. Ich meine den Min- oder Max-Wert selbst.

Reply:
das Problem ist, das ich ja eine group habe in der min und max liegen. An die komme ich nicht direkt dran über einen Namen. Der Validator liegt auf der Ebene des forms. Deshalb so.